Update links to `cardano-wallet` repository.

jonathanknowles opened the pull request:
## Issue

None.

## Details

This PR updates links to point to the new location of the `cardano-wallet` repository within `cardano-foundation`. It applies the following search and replace operation across the whole repository:

```patch
- https://github.com/input-output-hk/cardano-wallet
+ https://github.com/cardano-foundation/cardano-wallet
```

jonathanknowles committed:
Update references to `cardano-wallet` repository in `hs` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `md` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `sh` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `yml` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `cabal` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `nix` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `yaml` files.

jonathanknowles committed:
Update references to `cardano-wallet` repository in `Dockerfile` files.

HeinrichApfelmus commented on .buildkite/check-bors.sh:
I suggest to remove this file in a separate PR.

jonathanknowles replied:
> I suggest to remove this file in a separate PR.

Good point. I'll make a separate PR for this.

HeinrichApfelmus commented on docs/site/src/contributor/how/release-checklist.md:
```diff
@@ -4,7 +4,7 @@
 ## Check additional automated tests
 - [ ] Verify that the latest E2E tests are fine (See "E2E *" workflows on https://github.com/cardano-foundation/cardano-wallet/actions).
-- [ ] Verify latest [buildkite nightly](https://buildkite.com/input-output-hk/cardano-wallet-nightly) and make sure the results are fine. [Benchmark charts](http://ec2-44-197-192-237.compute-1.amazonaws.com:5555) may be helpful in analysis.
+- [ ] Verify latest [buildkite nightly](https://buildkite.com/cardano-foundation/cardano-wallet-nightly) and make sure the results are fine. [Benchmark charts](http://ec2-44-197-192-237.compute-1.amazonaws.com:5555) may be helpful in analysis.
```
The Buildkite organization is currently called `cardanofoundation`. I will change it to `cardano-foundation`, so that the link becomes correct.

Note: This change contradicts the pull request description. 🤓😈

jonathanknowles replied:
> Note: This change contradicts the pull request description.

Ah, thanks for spotting this!

My preference would have been to exclude this change from the PR, since as you point out, it's not within this PR's scope.

But given that you're happy to change the Buildkite organisation name from `cardanofoundation` to `cardano-foundation`, I'm happy to leave this change in!

HeinrichApfelmus replied:
Oops. Turns out that I cannot change the name because there already exists a Buildkite organization `cardano-foundation`. 🙈

HeinrichApfelmus replied:
Fixed.

HeinrichApfelmus reviewed (approved):
Looks good to me! Modulo changes.

jonathanknowles committed:
Fix outdated link in `Cardano.Wallet.Address.DerivationSpec`.

jonathanknowles committed:
Update to a newer version of `cardano-addresses`.

The newer version of `cardano-addresses` includes an updated link to the
new location of the `cardano-wallet` repository. Without this update,
the following unit tests for `cardano-wallet` will fail:

"/Cardano.Wallet.Api/Malformed BodyParam/(WalletOrAccountPostData) POST /wallets/"
"/Cardano.Wallet.Api/Malformed BodyParam/(WalletPutPassphraseData) PUT /wallets/0000000000000000000000000000000000000000/passphrase/"
"/Cardano.Wallet.Api/Malformed BodyParam/(SomeByronWalletPostData) POST /byron-wallets/"
